src/client/Microsoft.Identity.Client/Kerberos/KerberosSupplementalTicketManager.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Json.Linq;
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+
+namespace Microsoft.Identity.Client.Kerberos
+{
+    /// <summary>
+    /// Helper class to manage Kerberos Ticket Claims.
+    /// </summary>
+    public static class KerberosSupplementalTicketManager
+    {
+        private const int DefaultLogonId = 0;
+        private const string KerberosClaimType = "xms_as_rep";
+        private const string IdTokenAsRepTemplate = @"{{""id_token"": {{ ""xms_as_rep"":{{""essential"":""false"",""value"":""{0}""}} }} }}";
+        private const string AccessTokenAsRepTemplate = @"{{""access_token"": {{ ""xms_as_rep"":{{""essential"":""false"",""value"":""{0}""}} }} }}";
+
+        /// <summary>
+        /// Creates a <see cref="KerberosSupplementalTicket"/> object from given ID token string..
+        /// </summary>
+        /// <param name="idToken">ID token string.</param>
+        /// <returns>A <see cref="KerberosSupplementalTicket"/> object if exists and parsed correctly. Null, otherwise.</returns>
+        public static KerberosSupplementalTicket FromIdToken(string idToken)
+        {
+            if (string.IsNullOrEmpty(idToken) || idToken.Length < 128)
+            {
+                // Token is empty or too short -ignore parsing.
+                return null;
+            }
+
+            KerberosIdTokenParser jwt = KerberosIdTokenParser.Parse(idToken);
+            if (jwt == null)
+            {
+                return null;
+            }
+
+            string kerberosAsRep = jwt.GetValueOrEmptyString(KerberosClaimType);
+            if (string.IsNullOrEmpty(kerberosAsRep))
+            {
+                return null;
+            }
+
+            return (KerberosSupplementalTicket)JsonConvert.DeserializeObject(kerberosAsRep, typeof(KerberosSupplementalTicket));
+        }
+
+        /// <summary>
+        /// Save current Kerberos Ticket to current user's Ticket Cache.
+        /// </summary>
+        /// <param name="ticket">Kerberos ticket object to save.</param>
+        /// <remarks>Can throws <see cref="ArgumentException"/> when given ticket parameter is not a valid Kerberos Supplemental Ticket.
+        /// Can throws <see cref="Win32Exception"/> if error occurs while saving ticket information into Ticket Cache.
+        /// </remarks>
+        public static void SaveToCache(KerberosSupplementalTicket ticket)

```

Comment:
We might want to be more precise than "Cache", as this is a different cache from the token cache for instance. What about `SaveToWindowsTicketCache` ?

Reply:
Renamed.